pkg/controller/scyllacluster/resource.go

```diff
@@ -882,6 +925,20 @@ func stringOrDefault(str, def string) string {
 	return def
 }
 
+func ptrValueOrDefault[T any](v *T, def T) T {
+	if v != nil {
+		return *v
+	}
+	return def
+}
+
+func copyPtr[T any](v *T) *T {
```

Comment:
Not a fan of the name of this func, the name implies the function returns a copy of the address?

Reply:
changed the name

Reply:
I think `dupPtr` has the same issue. Idk, maybe just `copyValue`, `copyReferencedValue`? The fact that it returns a pointer doesn't need to be reflected in the name imo, but what it does internally (i.e. copy a dereferenced value) should.

Reply:
changed to `copyReferencedValue`